src/lib.rs

```diff
@@ -50,46 +50,55 @@ pub fn derive_deref_mut(input: TokenStream) -> TokenStream {
 fn parse_fields(item: &syn::DeriveInput, mutable: bool) -> (syn::Type, proc_macro2::TokenStream) {
     let trait_name = if mutable { "DerefMut" } else { "Deref" };
     let fields = match item.data {
-        syn::Data::Struct(ref body) => body.fields.iter().collect::<Vec<&syn::Field>>(),
+        syn::Data::Struct(ref body) =>
+            body.fields.iter()
+                .filter(|field| {
+                    if let Type::Path(TypePath { path: Path { segments, .. }, .. }) = &field.ty {
+                        if segments
+                            .last()
+                            .expect("Expected path to have at least one segment")
+                            .ident == "PhantomData"
```

Comment:
This whole thing is hard for me to parse. Can we move the left side to a local variable?